src/main/java/randoop/generation/GrtImpurity.java

```diff
@@ -0,0 +1,513 @@
+
+    // Do not fuzz void, char, boolean, or byte.
+    if (outputType.isVoid()
+        || outputType.runtimeClassIs(char.class)
+        || outputType.runtimeClassIs(Character.class)
+        || outputType.runtimeClassIs(boolean.class)
+        || outputType.runtimeClassIs(Boolean.class)
+        || outputType.runtimeClassIs(byte.class)
+        || outputType.runtimeClassIs(Byte.class)) {
+      return new GrtImpurityAndNumStatements(sequence, 0);
+    }
+
+    Class<?> outputClass = outputType.getRuntimeClass();
+    List<Executable> fuzzingOperations = new ArrayList<>();
+
+    // Append statements for fuzzing operations based on the type of the output object,
+    // then get the fuzzing operations.
+    try {
+      if (outputClass.isPrimitive()) { // fuzzing primitive numbers
+        sequence = appendGaussianSampleSequence(sequence, outputClass);
+        fuzzingOperations = getNumberFuzzingMethods(outputClass);
+      } else if (outputClass == String.class) { // fuzzing String
+        // Randomly select a fuzzing operation for String.
+        StringFuzzingOperation operation =
+            StringFuzzingOperation.values()[
+                Randomness.nextRandomInt(StringFuzzingOperation.values().length)];
+        sequence = appendStringFuzzingInputs(sequence, operation, fuzzStatementOffset);
+        if (fuzzStatementOffset.offset == 0) { // sequence not fuzzed, return original sequence
+          return new GrtImpurityAndNumStatements(sequence, 0);
+        }
+        fuzzingOperations = getStringFuzzingMethod(operation);
+      } else {
+        // TODO: Fuzz other objects based on purity analysis
+        //  return the original sequence for now
+        return new GrtImpurityAndNumStatements(sequence, 0);
+      }
+    } catch (Exception e) { // All other exceptions are unexpected
+      throw new RandoopBug("GRT Impurity fuzzing failed: " + e.getMessage(), e);
+    }
+
+    Sequence output = sequence;
+
+    // TODO: Implement cast to improve short fuzzing output readability
+    //  (e.g. cast to Integer for Integer.sum when fuzzing short can make Integer.valueOf()
+    //  unnecessary).
+
+    // Append fuzzing operations to the output sequence.
+    Iterator<Executable> iterator = fuzzingOperations.iterator();
+    while (iterator.hasNext()) {
+      Executable executable = iterator.next();
+      output =
+          appendFuzzingOperation(
+              output, executable, getOutputType(executable), fuzzStatementOffset);
+    }
+
+    return new GrtImpurityAndNumStatements(output, fuzzStatementOffset.offset);
+  }
+
+  /**
+   * Create a new sequence with a fuzzing operation statement appended to the given sequence.
+   *
+   * @param sequence the sequence to append the fuzzing operations to
+   * @param fuzzingOperation the method to be invoked to fuzz the object
+   * @param outputType the output type of the fuzzing operation
+   * @param fuzzStatementOffset the offset counter for the number of fuzzing statements added
```

Comment:
Explain the semantics of "offset counter", or drop that part of the comment and just say "the number of fuzzing statements added".
Is this the number of fuzzing statements added so far?  Added ever?  Is it side-effected by this method?